var/spack/repos/builtin/packages/py-rpy2/package.py

```diff
@@ -10,39 +10,43 @@ class PyRpy2(PythonPackage):
     interface to R from Python, a proposed high-level interface,
     including wrappers to graphical libraries, as well as R-like
     structures and functions.
-
     """
 
+    homepage = "https://rpy2.github.io"
     pypi = "rpy2/rpy2-2.5.4.tar.gz"
 
     license("GPL-2.0-or-later")
 
+    maintainers("Chrismarsh")
+
+    version("3.5.17", sha256="dbff08c30f3d79161922623858a5b3b68a3fba8ee1747d6af41bc4ba68f3d582")
     version("3.0.4", sha256="2af5158a5d56af7f7bf5e54d8d7e87b6f115ff40f056d82f93cad0cbf6acc0cb")
     version("3.0.0", sha256="34efc2935d9015527837d6b1de29641863d184b19d39ad415d5384be8a015bce")
```

Comment:
Does 2.9.4 not work at all with the lowest supported python version in spack (3.8)? Can we keep these legacy versions and deprecate them first before removal?
```suggestion
    version("3.0.0", sha256="34efc2935d9015527837d6b1de29641863d184b19d39ad415d5384be8a015bce")
    with default_args(deprecated=True):
```

Reply:
I don't know. The older versions (e.g., 2.5.6) `setup.py` doesn't have the version checks on python that the newer `setup.py` has. `rpy2@2.5.4` is from 2014, which significantly predates python 3.8. My initial reaction was that maintaining a set of previous major versions from 10 years ago wasn't worth the effort.

Reply:
Looking through these older versions, there are a bunch of extra version constraints that will need to be added. I've deprecated all the older 3.x versions, and added the requirements to support back to python 3.9